### PR TITLE
fix: optimised ci trigger rbac, updated handling for invalid roles in user creation request

### PR DESCRIPTION
**Pull request by kartik-579**
# Description
Optimised ci trigger rbac enforcement logic. Added handling for invalid role in user creation/updation request.

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/appStore/InstalledAppRestHandler.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (handler InstalledAppRestHandlerImpl) GetAllInstalledApp(w http.ResponseWri
 	token := r.Header.Get("token")
 	userEmailId, err := handler.userAuthService.GetEmailFromToken(token)
 	if err != nil {
-		handler.Logger.Errorw("error in getting user emailId from token", "userId", userId, "token", token)
+		handler.Logger.Errorw("error in getting user emailId from token", "userId", userId)
```
Comment (kripanshdevtron) [inline api/appStore/InstalledAppRestHandler.go:110]: error is not logged

File: api/cluster/EnvironmentRestHandler.go [hunk 2]
```diff
@@ -356,7 +356,7 @@ func (impl EnvironmentRestHandlerImpl) GetCombinedEnvironmentListForDropDown(w h
 	token := r.Header.Get("token")
 	userEmailId, err := impl.userService.GetEmailFromToken(token)
 	if err != nil {
-		impl.logger.Errorw("error in getting user emailId from token", "userId", userId, "token", token)
+		impl.logger.Errorw("error in getting user emailId from token", "userId", userId)
```
Comment (kripanshdevtron) [inline api/cluster/EnvironmentRestHandler.go:359]: error is not logged

File: api/appStore/InstalledAppRestHandler.go (hunk 1 above)
Comment (kartik-579) [reply api/appStore/InstalledAppRestHandler.go:110]: done

File: api/cluster/EnvironmentRestHandler.go (hunk 2 above)
Comment (kartik-579) [reply api/cluster/EnvironmentRestHandler.go]: done

**Pull request merged**